### PR TITLE
Update HTML minimzier to use ANTLR tokenizer 

### PR DESCRIPTION
**Pull request by mpherman2**
Small changes to the HTML minimizer to use the new tokenizer. I also added unit tests. Everything in the tokenizer/grammars dir is auto-generated.

### CODE REVIEW

File: src/python/tests/core/bot/minimizer/html_minimizer_test.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    helpers.patch(
+        self,
+        [('js_minimizer', 'bot.minimizer.js_minimizer.JSMinimizer.minimize'),
+         ('html_minimizer',
+          'bot.minimizer.chunk_minimizer.ChunkMinimizer.minimize'),
+         ('line_minimizer',
+          'bot.minimizer.delta_minimizer.DeltaMinimizer.minimize')])
+
+    self.mock.js_minimizer.side_effect = self._mock_js_minimization
+    self.mock.html_minimizer.side_effect = self._mock_html_minimization
```
[the remark this answers is not in the document]
Comment (mpherman2) [reply src/python/tests/core/bot/minimizer/html_minimizer_test.py]: I am still inexperienced with unit testing, so let me know if this is not the best way. My understanding is that It's too slow to run the full minimization. The HTML Minimizer's job is just to pass the data to a series of sub-minimizers. This test should just makes sure that that happens correctly and the results are combined in the right way. JS Minimizer is already being tested in its own unit tests. I don't believe tests were written for Chunk and Delta minimizers, but should probably be added eventually.

[the remark this answers is not in the document]
Comment (mpherman2) [reply src/python/tests/core/bot/minimizer/html_minimizer_test.py]: I added a comment to explain

File: src/python/tests/core/bot/minimizer/html_minimizer_test.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  def _mock_js_minimization(self, minimizer, data):
+    """Mock js_minimization. Returns simple js_minimized code."""
+    # pylint: disable=unused-argument
+    return minimizer.token_combiner(
+        ['var b = document.createTextNode("PASS")\n'])
+
+  def _mock_html_minimization(self, minimizer, data):
+    """Mock html minimization. Returns minimized top tags when its called for
+    the first section, otherwise returns minimized bottom section"""
+    #The first 3 times the html_minimizer is called will be for the begining.
```
[the remark this answers is not in the document]
Comment (mpherman2) [reply src/python/tests/core/bot/minimizer/html_minimizer_test.py]: Done

File: src/python/tests/core/bot/minimizer/html_minimizer_test.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Dor1s) [reply src/python/tests/core/bot/minimizer/html_minimizer_test.py]: Thanks, this makes sense, and the comment is good to have!

**Pull request merged**